hollow/src/main/java/com/netflix/hollow/core/write/HollowWriteStateEngine.java

```diff
@@ -245,6 +247,28 @@ public void addAllObjectsFromPreviousCycle() {
         }
     }
 
+    /**
+     * Add all objects of the provided type from the previous cycle, exactly as they were in the previous cycle, while
+     * also considering referenced records.
+     * @param typeName The type name of the records that should added from the previous cycle
+     * @param readStateEngine The read state of the previous cycle (should be retrieved using the method linked below)
+     * {@link com.netflix.hollow.api.producer.HollowProducer.WriteState#getPriorState()}
+     */
+    public void addAllObjectsFromPreviousCycle(String typeName, HollowReadStateEngine readStateEngine) {
```

Comment:
As you point out, this is what `TransitiveSetTraverser` is for. There's a few reasons I'm against adding this helper method:

* your use use-cases is one of several possible use cases. I'm not confident it's the primary use-case, for example  the incremental producer does something similar with `TransitiveSetTraverser`

* I want to avoid bloating `HollowWriteStateEngine` with helper methods, especially per above if there are a number of slight variations for which a helper method would be justified

* The contract and the API are inconsistent. Nothing prevents users from using an arbitrary HollowReadStateEngine as the second arguments. More likely to cause confusion than be helpful.

Reply:
@toolbear I totally agree with your reasons, but don't you think it's useful to have the capability of independently publishing the datasets somewhere in the producer? I think that restoring a dataset with reference-type awareness should be an internal detail of hollow, and not something the the user should be aware of.